### PR TITLE
Improved validation and execution for web services

### PR DESCRIPTION
**Pull request by hypeJunction**
[ ] squash?
[ ] update tests

### CODE REVIEW

**Comment by hypeJunction** [comment]
Sorry for the white space noise. Take a look if this is something we can pull into core, and I will clean up the noise.

**Comment by juho-jaakkola** [comment]
Fine with me

File: mod/web_services/lib/web_services.php [hunk 1]
```diff
@@ -147,19 +141,45 @@ function get_parameters_for_method($method) {
...
+	}
+
+	foreach ($expected_parameters as $key => $spec) {
+		// Make things go through the sanitiser
+		$default = elgg_extract('default', $spec);
+		$value = get_input($key, $default);
+
+		// Cast values to specifid type
+		$type = elgg_extract('type', $spec);
+		switch ($type) {
```
Comment (juho-jaakkola) [inline mod/web_services/lib/web_services.php]: Perhaps the switch statement could be replaced with ```settype()```?

**Comment by hypeJunction** [comment]
Will do some WS revamping for 2.0 and include these there.

File: mod/web_services/lib/web_services.php (hunk 1 above)
Comment (hypeJunction) [inline mod/web_services/lib/web_services.php]: Yes, sounds nice. Just need to normalize types. I think we need to add type normalization to ```elgg_ws_expose_function()```

Comment (hypeJunction) [inline mod/web_services/lib/web_services.php]: Never minds, PHP docs say we can use either since 4.2.0

**Pull request closed without merging**